Classes/Parser/ScssParser.php

```diff
@@ -83,25 +84,88 @@ protected function parseFile($file, $settings)
                 'sourceMapBasepath' => '<PATH DOES NOT EXIST BUT SUPRESSES WARNINGS>'
             ]);
         }
-        $css = $scss->compile('@import "' . $file . '"');
+
+        $css = $scss->compile($this->resolveCode($file));
 
         $absoluteFilename = GeneralUtility::getFileAbsFileName($file);
         $relativePath = $settings['cache']['tempDirectoryRelativeToRoot'] . dirname(substr($absoluteFilename, strlen($this->getPathSite()))) . '/';
         $search = '%url\s*\(\s*[\\\'"]?(?!(((?:https?:)?\/\/)|(?:data:?:)))([^\\\'")]+)[\\\'"]?\s*\)%';
         $replace = 'url("' . $relativePath . '$3")';
         $css = preg_replace($search, $replace, $css);
 
+        $parsedFiles = $scss->getParsedFiles();
+        if (!isset($parsedFiles[$absoluteFilename])) {
+            $parsedFiles[$absoluteFilename] = filemtime($absoluteFilename);
+        }
+
         return [
             'css' => $css,
             'cache' => [
                 'version' => Version::VERSION,
                 'date' => date('r'),
                 'css' => $css,
                 'etag' => md5($css),
-                'files' => $scss->getParsedFiles(),
+                'files' => $parsedFiles,
                 'variables' => $settings['variables'],
                 'sourceMap' => $settings['options']['sourceMap']
             ]
         ];
     }
+
+    /**
+     * Substitutes relative @import paths with according matching visual paths.
+     * This scenarios happens, when e.g. developing packages using the `path`
+     * repository feature of Composer - having one package in `public/typo3conf/ext/`
+     * and the other one symlinked in e.g. `packages/`.
+     *
+     * Since the SCSS parser works on resolved real paths, the symlinked context
+     * is lost. This method tries to keep the meaning for resolving absolute
+     * references to their matching directories.
+     *
+     * @param string $file
+     * @return string
+     */
+    protected function resolveCode($file)
+    {
+        $visualFilePath = GeneralUtility::getFileAbsFileName($file);
+        $realFilePath = realpath($visualFilePath);
+        if ($visualFilePath === $realFilePath) {
+            return '@import "' . $file . '"';
+        }
+
+        $visualDirectory = dirname($visualFilePath);
+        $realDirectory = dirname($realFilePath);
+        $code = file_get_contents($visualFilePath);
+        if (preg_match_all('#@import\s+"([^"]+)"#', $code, $matches)) {
```

Comment:
If i understand it correctly, the issues can only occur in one file, which is the entry point file for parsing, which can reference other scss files on other extensions.

Two suggestions how this can be resolved cleanly:

1. Support `EXT:extension_key` syntax and replace all occurrences with absolute paths (similar to the approach here, but a bit more predictable).

2. Always use absolute fully qualified paths with a stream wrapper URI like that: `typo3://ext_key/Resources/...` and register a stream wrapper that can resolve these to absolute paths. Have not checked that, but the SCSS parser should be able to handle such paths. Benefit: no parsing needs to be done in the source files and any file can reference other files from extensions that way easily.

Reply:
> If i understand it correctly, the issues can only occur in one file, which is the entry point file for parsing, which can reference other scss files on other extensions.

Basically it can happen in the whole `@import` chain and every time paths to other components are defined using path-traversals (`../../../whatever`).

on 1) okay, needs additional adjustments in generated code (e.g. created from https://www.sitepackagebuilder.com/ - that was my use-case)
* however this (still) only works on the caller side, since 3rd party library does not know anything about `EXT:...`
* (it feels like the "crash" @benjaminkott referred to is caused by pre-parsing and substituting paths during runtime - the patch does not have memory management nor detection for infinite loops)

on 2) the cleanest solution but also the one that cannot be used "right now"
* requires adjustments on existing (generated) code (see previous remark)
* `typo3://` (or any other stream wrapper) should be defined carefully in order to avoid confusion - this would be something for TYPO3 core and not only for the Bootstrap Package parsing sub component

Reply:
> the cleanest solution

Glad we have an agreement here 👍

> but also the one that cannot be used "right now"

True. However, when we agree and join forces, it could be used in a very near future (closer than "another year").

> typo3:// (or any other stream wrapper) should be defined carefully in order to avoid confusion

I agree.

> this would be something for TYPO3 core and not only for the Bootstrap Package parsing sub component

Well, `typo3://` was just an example. Could be anything and the wrapper could be registered only for parsing and removed after parsing to lower conflict possibilities. E.g. `scssfiles://foo/main.scss` which resolves to `typo3conf/ext/foo/Resources/Private/Styles/Scss/main.scss`, or even cleaner, introduce a registry for path mappings.

I'd contribute the code for this package here, but only if it is accepted as solution to avoid working for the bin.